### ISSUE TITLE
Something is wrong with ami_l1b reader

### ISSUE THREAD

**Issue by yukaribbba** [pytroll/satpy#2860]
**Describe the bug**
The output image is not right. Below is the VI006 band. I've checked via GDAL and the dataset is not damaged.
![image](https://github.com/user-attachments/assets/9f3bf572-9000-4a48-a487-172566c63514)

**To Reproduce**

```python
files = find_files_and_readers(base_dir="C:/Users/45107/Downloads/Sat/Geo/gk2a",
                               reader="ami_l1b",
                               )
scn = Scene(filenames=files)
product = "VI006"
scn.load([product])
scn.show(product)

```

**Actual results**
```python
[DEBUG: 2024-07-20 16:54:57 : satpy.readers.yaml_reader] Reading ('C:\\Users\\45107\\miniforge3\\Lib\\site-packages\\satpy\\etc\\readers\\ami_l1b.yaml',)
[DEBUG: 2024-07-20 16:54:57 : satpy.readers.yaml_reader] Reading ('C:\\Users\\45107\\miniforge3\\Lib\\site-packages\\satpy\\etc\\readers\\ami_l1b.yaml',)
[DEBUG: 2024-07-20 16:54:57 : satpy.readers.yaml_reader] Assigning to ami_l1b: ['C:/Users/45107/Downloads/Sat/Geo/gk2a\\gk2a_ami_le1b_vi004_fd010ge_202407190240.nc', 'C:/Users/45107/Downloads/Sat/Geo/gk2a\\gk2a_ami_le1b_vi005_fd010ge_202407190240.nc', 'C:/Users/45107/Downloads/Sat/Geo/gk2a\\gk2a_ami_le1b_vi006_fd005ge_202407190240.nc', 'C:/Users/45107/Downloads/Sat/Geo/gk2a\\gk2a_ami_le1b_vi008_fd010ge_202407190240.nc']
[DEBUG: 2024-07-20 16:54:57 : rasterio.env] PROJ data files are available at built-in paths.
[DEBUG: 2024-07-20 16:54:58 : satpy.composites.config_loader] Looking for composites config file ami.yaml
[DEBUG: 2024-07-20 16:54:58 : satpy.composites.config_loader] Looking for composites config file visir.yaml
[DEBUG: 2024-07-20 16:54:58 : pyorbital.tlefile] Path to the Pyorbital configuration (where e.g. platforms.txt is found): C:\Users\45107\miniforge3\Lib\site-packages\pyorbital\etc
C:\Users\45107\miniforge3\Lib\site-packages\satpy\readers\ami_l1b.py:200: RuntimeWarning: overflow encountered in scalar subtract
  data &= 2**bits - 1
[DEBUG: 2024-07-20 16:54:58 : satpy.writers] Adding enhancement configuration from file: C:\Users\45107\miniforge3\Lib\site-packages\satpy\etc\enhancements\generic.yaml
[DEBUG: 2024-07-20 16:54:58 : satpy.writers] Data for DataID(name='VI006', wavelength=WavelengthRange(min=0.599, central=0.639, max=0.679, unit='µm'), resolution=500, calibration=<1>, modifiers=()) will be enhanced with options:
	[{'name': 'linear_stretch', 'method': <function stretch at 0x000001A6E23027A0>, 'kwargs': {'stretch': 'crude', 'min_stretch': 0.0, 'max_stretch': 100.0}}, {'name': 'gamma', 'method': <function gamma at 0x000001A6E2302660>, 'kwargs': {'gamma': 1.5}}]
[DEBUG: 2024-07-20 16:54:58 : trollimage.xrimage] Applying stretch crude with parameters {'min_stretch': 0.0, 'max_stretch': 100.0}
[DEBUG: 2024-07-20 16:54:58 : trollimage.xrimage] Applying gamma 1.5

```

**Environment Info:**
 - Satpy Version: 0.49
 - Xarray Version: 2024.6.0
 - Dask Version: 2024.7.1

**Comment by simonrp84** [comment]
Hi, I fixed this yesterday in #2806. It's a problem with compatibility with the latest version of numpy. You can either use `main`, which has the fix merged, or downgrade numpy to 1.26 until a new satpy release is made.

**Comment by yukaribbba** [comment]
Ah I was too busy to catch up with recent updates...It's fine now. Thanks!
